src/cli/cmd_build_and_check.rs

```diff
@@ -265,34 +274,59 @@ pub fn cmd_test(args: CmdTest) -> miette::Result<()> {
 
     // compile
     let mut sources = Sources::new();
-    let mut tast = TypeChecker::new();
-    let _node_id = typecheck_next_file(
-        &mut tast,
-        None,
-        &mut sources,
-        args.path.to_string(),
-        code,
-        0,
-    )?;
 
-    let kimchi_vesta = KimchiVesta::new(!args.no_double);
-    let compiled_circuit = compile(&sources, tast, kimchi_vesta)?;
-
-    let (prover_index, verifier_index) = compiled_circuit.compile_to_indexes()?;
-    println!("successfully compiled");
-
-    // print ASM
-    let asm = prover_index.asm(&sources, args.debug);
-    println!("{asm}");
+    match backend.as_str() {
+        "kimchi-vesta" => {
+            let mut tast = TypeChecker::new();
+            let _node_id = typecheck_next_file(
```

Comment:
I'm always confused when I look at this code, where we have to init Sources, then init typechecker, then call typecheck_next_file, and then I'm wondering which one do we call if we have several files :D

I'm not sure what would be a good alternative, but we should probably create an issue to brainstorm that. First, I think we should have convenience functions in the compiler for any usecase (compile a single file, compile a single file from source, compile multiple files from source, etc.) 

second, maybe we could have some sort of builder pattern to make the flow much clearer. Anyway it's not super high priority but my brain always thinks of cleaning up ^^

Reply:
That would be better: https://github.com/zksecurity/noname/issues/67